r/tests/testthat/test-Table.R

```diff
@@ -145,7 +145,55 @@ test_that("[, [[, $ for Table", {
   expect_data_frame(tab[0], tbl[0])
 })
 
+test_that("[[<- assignment", {
+  tbl <- tibble::tibble(
+    int = 1:10,
+    dbl = as.numeric(1:10),
+    lgl = sample(c(TRUE, FALSE, NA), 10, replace = TRUE),
+    chr = letters[1:10],
+    fct = factor(letters[1:10])
+  )
+  tab <- Table$create(tbl)
+
+  # can remove a column
+  tab[["chr"]] <- NULL
+  expect_data_frame(tab, tbl[-4])
+
+  # can add a named column
+  tab[["new"]] <- letters[10:1]
+  expect_vector(tab[["new"]], letters[10:1])
```

Comment:
Part of the behavior we want to test is the position of the new column, so let's assert the whole data.frame

```suggestion
  expect_data_frame(tab, cbind(tbl, new = letters[10:1]))
```